File: time_manager.py
import datetime

# időkezelés
class TimeManager:
    def __init__(self):
        self.start_time = datetime.datetime.now()
        self.last_check = self.start_time

    def get_current_time(self):
        return datetime.datetime.now()

    # Visszaadja az eltelt időt a legutóbbi ellenőrzés óta.
    def get_elapsed_time(self):
        now = self.get_current_time()
        elapsed = now - self.last_check
        self.last_check = now
        return elapsed

    # Visszaadja az összes eltelt időt a kezdet óta.
    def get_total_elapsed_time(self):
        return datetime.datetime.now() - self.start_time

File: loligochi/loligochi-classlib/TimeManager.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace loligochi_classlib
{
    public class TimeManager
    {
        public DateTime StartTime { get; set; }
        public DateTime LastCheck { get; set; }

        public TimeManager()
        {
            StartTime = DateTime.Now;
            LastCheck = StartTime;
        }

        public DateTime GetCurrentTime()
        {
            return DateTime.Now;
        }

        public TimeSpan GetTheTimeThatElapsedTillWeStartedTheGame()
        {
            DateTime now = GetCurrentTime();
            TimeSpan elapsed = now - StartTime;
            return elapsed;
        }

        // Returns the total elapsed time since the start.
        public TimeSpan GetTotalElapsedTime()
        {
            return DateTime.Now - StartTime;
        }
    }
}
